plugins/test/fu-plugin-test-fail.c

```diff
@@ -0,0 +1,39 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+ *
+ * Licensed under the GNU General Public License Version 2
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "config.h"
+
+#include "fu-plugin.h"
+#include "fu-plugin-vfuncs.h"
+
+gboolean
+fu_plugin_update (FuPlugin *plugin,
```

Comment:
I don't think you need a new plugin for such a simple method.  Why not just use the existing plugin and key off of a bad payload?

Reply:
I tried this, but the other plugin is specifically testing the registered callback handling which would be a PITA to set up in these test cases. Maybe I could disable the coldplug registration stuff, although I'm not sure an env var is going to work.

Reply:
I think an env variable will work and without much extra code added too.  Rather than looking just for the presence of FWUPD_TESTS env variable, look at what that's set to.

If it's set to FWUPD_TESTS="registration" do the current one.  If it's FWUPD_TESTS="fail", set it up for fail on update.

Reply:
Okay, will try that.